core/ipc/rpc/camel-impl/src/main/java/org/opennms/core/rpc/camel/CamelRpcServerRouteManager.java

```diff
@@ -77,7 +76,11 @@ private DynamicRpcRouteBuilder(CamelContext context, MinionIdentity identity, Rp
         }
 
         public String getRouteId() {
-            return "RPC.Server." + module.getId();
+            return "RPC.Server." + module.getId() + "-" + Math.abs(module.hashCode());
```

Comment:
I put this in because the DNS RPC service was being added twice in some of the tests causing tests to flap (depending on whether or not the second service initialized before the Camel context started). By making the routes unique, you can see that two modules are trying to register for the same RPC service and trace that back to the log messages that I added to bind/unbind.